event.c

```diff
@@ -627,9 +629,54 @@ static void event_ffmpeg_newfile(struct context *cnt,
      *  motion movies get the same name as normal movies plus an appended 'm'
      *  PATH_MAX - 4 to allow for .mpg to be appended without overflow
      */
-    snprintf(cnt->motionfilename, PATH_MAX - 4, "%s/%sm", cnt->conf.filepath, stamp);
-    snprintf(cnt->newfilename, PATH_MAX - 4, "%s/%s", cnt->conf.filepath, stamp);
-
+     /* Set up a testing / evaluation codec which will use a different
+      * codec for the events.
+     */
+    snprintf(codec, sizeof(codec), "%s", cnt->conf.ffmpeg_video_codec);
+    if (strcmp(codec, "test") == 0) {
+        MOTION_LOG(NTC, TYPE_ENCODER, NO_ERRNO, "%s Running test of the various output formats.");
+        codenbr = cnt->event_nr;
+        while (codenbr > 10){
+          codenbr -= 10;
+        }
+        switch (codenbr) {
+        case 1:
+            snprintf(codec, sizeof(codec), "%s", "mpeg4");
```

Comment:
This code seems quite verbose; unless I've missed something I think "codec" could be declared as a const char *. Then this snprintf line can become simply codec = "mpeg4"; (and the line before the switch can be codec = cnt->conf.ffmpeg_video_codec)